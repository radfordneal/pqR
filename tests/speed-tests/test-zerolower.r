source("time.r")
 
test.name <- "zerolower.1"

zerolower1 <- test.cmp (function (A,n)
{ for (k in 1:n)
    for (j in 1:min(nrow(A),ncol(A)))
      A[j:nrow(A),j] <- 0
  A
})

A <- matrix(1,7,10)
sys.time(B <- zerolower1(A,50000))
print(B)

A <- matrix(1,52,50)
sys.time(B <- zerolower1(A,5000))

test.name <- "zerolower.2"

zerolower2 <- test.cmp (function (A,n)
{ nr <- nrow(A)
  nc <- ncol(A)
  for (k in 1:n)
    for (j in 1:min(nr,nc))
      A[j:nr,j] <- 0
  A
})

A <- matrix(1,7,10)
sys.time(B <- zerolower2(A,100000))
print(B)

A <- matrix(1,52,50)
sys.time(B <- zerolower2(A,10000))

test.name <- "zerolower.3"

zerolower3 <- test.cmp (function (A,n)
{ nr <- nrow(A)
  nc <- ncol(A)
  for (k in 1:n)
    for (j in 1:min(nr,nc))
      for (i in j:nr)
        A[i,j] <- 0
  A
})


A <- matrix(1,7,10)
sys.time(B <- zerolower3(A,20000))
print(B)

A <- matrix(1,52,50)
sys.time(B <- zerolower3(A,1000))
