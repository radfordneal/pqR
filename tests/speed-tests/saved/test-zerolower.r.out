
---- zerolower.1 

Time:
     [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
[1,]    0    1    1    1    1    1    1    1    1     1
[2,]    0    0    1    1    1    1    1    1    1     1
[3,]    0    0    0    1    1    1    1    1    1     1
[4,]    0    0    0    0    1    1    1    1    1     1
[5,]    0    0    0    0    0    1    1    1    1     1
[6,]    0    0    0    0    0    0    1    1    1     1
[7,]    0    0    0    0    0    0    0    1    1     1
Time:

---- zerolower.2 

Time:
     [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
[1,]    0    1    1    1    1    1    1    1    1     1
[2,]    0    0    1    1    1    1    1    1    1     1
[3,]    0    0    0    1    1    1    1    1    1     1
[4,]    0    0    0    0    1    1    1    1    1     1
[5,]    0    0    0    0    0    1    1    1    1     1
[6,]    0    0    0    0    0    0    1    1    1     1
[7,]    0    0    0    0    0    0    0    1    1     1
Time:

---- zerolower.3 

Time:
     [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
[1,]    0    1    1    1    1    1    1    1    1     1
[2,]    0    0    1    1    1    1    1    1    1     1
[3,]    0    0    0    1    1    1    1    1    1     1
[4,]    0    0    0    0    1    1    1    1    1     1
[5,]    0    0    0    0    0    1    1    1    1     1
[6,]    0    0    0    0    0    0    1    1    1     1
[7,]    0    0    0    0    0    0    0    1    1     1
Time:
